### PR TITLE
Add select all and deselect all buttons when exporting data

### PR DESCRIPTION
**Pull request by niushuai1991**
#18774 

printscreen 2023-08-22 014959

### CODE REVIEW

**Comment by niushuai1991** [comment]
I only added English and Chinese, because I am not familiar with others.

**Comment by LonwoLonwo** [comment]
Hello @niushuai1991 

Thanks for your contribution.

You helped me to find a bug. In your code and our code, the "Cancel" button will be ignored after clicking. It also works as the "Ok" button.
So, You need to think about the temporary state of actions list.

**Comment by niushuai1991** [comment]
@LonwoLonwo hi, After some tries, I changed the mappings to member variables and it's already working fine.

**Comment by niushuai1991** [comment]
> Hello @niushuai1991
> 
> Thanks for your contribution.
> 
> You helped me to find a bug. In your code and our code, the "Cancel" button will be ignored after clicking. It also works as the "Ok" button. So, You need to think about the temporary state of actions list.

I fixed this bug

**Comment by niushuai1991** [comment]
> Hello @niushuai1991
> 
> Thanks for your contribution.
> 
> You helped me to find a bug. In your code and our code, the "Cancel" button will be ignored after clicking. It also works as the "Ok" button. So, You need to think about the temporary state of actions list.

@LonwoLonwo OK, I fixed this bug.

**Review by sbOogway — APPROVED** [review]
clean

File: plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/stream/StreamConsumerPageSettings.java [hunk 1]
```diff
@@ -415,6 +421,35 @@ protected void setValue(Object element, Object value) {
                 column.getColumn().setText(DTUIMessages.stream_consumer_page_mapping_mapping_column_name);
             }
 
+            Composite panel = new Composite(group, SWT.NONE);
+            panel.setLayout(new FillLayout());
+            Button selectAllButton = new Button(panel, SWT.NONE);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/stream/StreamConsumerPageSettings.java]: Please consider using `UIUtils.createDialogButton` instead. They have padding and generally look better.

**Review by ShadelessFox — CHANGES_REQUESTED** [review]
There are too many changes IMO.

You can use the `refreshMappings` to acquire a copy of all mappings that you'll operate over in the dialog. After pressing the OK button, you can simply replace old mappings with new mappings.

**Comment by niushuai1991** [comment]
@ShadelessFox hello,Your suggestion is very helpful. I have optimized my code, please review it again.

**Review by ShadelessFox — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by DariaMarkaryan** [comment]
Verified

**Comment by ShadelessFox** [comment]
We decided that we wanted to allow changing mappings directly from the tree. Thanks for the contribution, regardless. https://github.com/dbeaver/dbeaver/pull/22409

**Pull request closed without merging**